python/lsst/daf/butler/_butler_collections.py

```diff
@@ -27,15 +27,73 @@
 
 from __future__ import annotations
 
-__all__ = ("ButlerCollections",)
+__all__ = ("ButlerCollections", "CollectionInfo")
 
 from abc import ABC, abstractmethod
-from collections.abc import Iterable
+from collections.abc import Iterable, Sequence, Set
+from typing import Any, overload
 
+from pydantic import BaseModel
 
-class ButlerCollections(ABC):
+from ._collection_type import CollectionType
+
+
+class CollectionInfo(BaseModel):
+    """Information about a single Butler collection."""
+
+    name: str
+    """Name of the collection."""
+    type: CollectionType
+    """Type of the collection."""
+    doc: str = ""
+    """Documentation string associated with this collection."""
+    children: tuple[str, ...] = tuple()
+    """Children of this collection (only if CHAINED)."""
+    parents: frozenset[str] | None = None
+    """Any parents of this collection.
+
+    `None` if the parents were not requested.
+    """
+    dataset_types: frozenset[str] | None = None
+    """Names of any dataset types associated with datasets in this collection.
+
+    `None` if no dataset type information was requested
+    """
+    governors: dict[str, frozenset[str]] | None = None
```

Comment:
@TallJimbo  is there a reason why governors should not be included since it comes "for free"?

Reply:
They just haven't proven very useful, except in the context of guessing default instrument values from collections at butler construction.

Reply:
Ok. I was thinking that it might be useful for immediately returning with no results and single reason (for the loop over dataset types case) if it was possible that after doing `results.where("instrument = 'HsC'")` it would be possible to ask the results object to return the governor dimensions involved in the query such that we could immediately say HsC isn't an instrument in this collection. Since we don't have that capability at the moment I will remove governors from the CollectionInfo for now and we can easily add it back later (the information is immediately available without an extra API call).

Reply:
Yes, that was my original thinking, too, and it's just that in practice I think the collection sequences we get passed have either the one instrument the user cares about, or none, or all of them, and hence the filter doesn't affect the query.